lib/index.ts

```diff
@@ -90,6 +91,29 @@ async function runInFat<T>(
 		});
 		fat.on('ready', resolve);
 	});
+	// Check whether fatfs added the promises namespace on their side
+	if (fat.promises == null) {
+		// Lazily populate the promise based variants
+		const originalFatKeys = Object.keys(fat);
+		Object.defineProperty(fat, 'promises', {
+			enumerable: true,
+			configurable: true,
+			get() {
+				const promises: Record<string, Function> = {};
```

Comment:
@thgreasi What about using 

```suggestion
				const promises: Record<string, (...args: any[]) => Promise<any>> = {};
```

For explicitly typing that the function returns a `Promise`? It has more explicit `any`  but I think it is better than the implicit ones you get with `Function`

Reply:
Sure :+1: